### PR TITLE
[RA2 Ch3] Scheduling pods with non-resilient apps

### LINKED ISSUE #2142: [RA2 Ch3] Add examples of pod scheduling requirements
To https://github.com/cntt-n/CNTT/blob/master/doc/ref_arch/kubernetes/chapters/chapter03.md#3.2.1 add examples of pod scheduling requirements for non-resilient applications.

### PR DESCRIPTION
**Pull request by petorre**
Fixes #2142 .

### CODE REVIEW

File: doc/ref_arch/kubernetes/chapters/chapter03.md [hunk 1]
```diff
@@ -206,6 +206,23 @@ which provides the isolation of Operating System kernels.
...
+
+<a name="3.2.1.8"></a>
+#### 3.2.1.8 Scheduling Pods with Non-resilient Applications
+
+Non-resilient applications are sensitive to platform impairments on Compute like pausing CPU cycles (for example because of OS scheduler) or Networking like packet drops, reordering or latencies. Such applications need to be carefully scheduled on nodes and preferably still decoupled from infrastructure details of those nodes.
+
+| # | Intensive on  | Not intensive on | Using hardware acceleration | Requirements for optimized pod scheduling |
+|---|---|---|---|---|
+| 1 | Compute | Networking (dataplane) | No | CPU Manager |
+| 2 | Compute | Networking (dataplane) | CPU instructions | #1 plus NFD |
```
Comment (tomkivlin) [inline doc/ref_arch/kubernetes/chapters/chapter03.md:218]:
```suggestion
| 2 | Compute | Networking (dataplane) | CPU instructions | CPU Manager, NFD |
```

File: doc/ref_arch/kubernetes/chapters/chapter03.md [hunk 2]
```diff
@@ -206,6 +206,23 @@ which provides the isolation of Operating System kernels.
...
+| 3 | Compute | Networking (dataplane) | Fixed function acceleration, Firmware-programmable network adapters or SmartNICs | #1 plus Device Plugin |
```
Comment (tomkivlin) [inline doc/ref_arch/kubernetes/chapters/chapter03.md:219]:
```suggestion
| 3 | Compute | Networking (dataplane) | Fixed function acceleration, Firmware-programmable network adapters or SmartNICs | CPU Manager, Device Plugin |
```

File: doc/ref_arch/kubernetes/chapters/chapter03.md [hunk 3]
```diff
@@ -206,6 +206,23 @@ which provides the isolation of Operating System kernels.
...
+| 4 | Networking (dataplane) | | No, or Fixed function acceleration, Firmware-programmable network adapters or SmartNICs  | Huge Pages (for DPDK-based applications), CPU Manager with configuration for isolcpus and SMT; Multiple interfaces; NUMA topology; Device Plugin |
+| 5 | Networking (dataplane) | | CPU instructions | #4 plus NFD |
```
Comment (tomkivlin) [inline doc/ref_arch/kubernetes/chapters/chapter03.md:221]:
```suggestion
| 5 | Networking (dataplane) | | CPU instructions | Huge Pages (for DPDK-based applications); CPU Manager with configuration for isolcpus and SMT; Multiple interfaces; NUMA topology; Device Plugin; NFD |
```

File: doc/ref_arch/kubernetes/chapters/chapter03.md [hunk 4]
```diff
@@ -206,6 +206,23 @@ which provides the isolation of Operating System kernels.
...
+#### 3.2.1.8 Scheduling Pods with Non-resilient Applications
+
+Non-resilient applications are sensitive to platform impairments on Compute like pausing CPU cycles (for example because of OS scheduler) or Networking like packet drops, reordering or latencies. Such applications need to be carefully scheduled on nodes and preferably still decoupled from infrastructure details of those nodes.
+
+| # | Intensive on  | Not intensive on | Using hardware acceleration | Requirements for optimized pod scheduling |
+|---|---|---|---|---|
+| 1 | Compute | Networking (dataplane) | No | CPU Manager |
+| 2 | Compute | Networking (dataplane) | CPU instructions | #1 plus NFD |
+| 3 | Compute | Networking (dataplane) | Fixed function acceleration, Firmware-programmable network adapters or SmartNICs | #1 plus Device Plugin |
+| 4 | Networking (dataplane) | | No, or Fixed function acceleration, Firmware-programmable network adapters or SmartNICs  | Huge Pages (for DPDK-based applications), CPU Manager with configuration for isolcpus and SMT; Multiple interfaces; NUMA topology; Device Plugin |
```
Comment (tomkivlin) [inline doc/ref_arch/kubernetes/chapters/chapter03.md:220]:
```suggestion
| 4 | Networking (dataplane) | | No, or Fixed function acceleration, Firmware-programmable network adapters or SmartNICs  | Huge Pages (for DPDK-based applications); CPU Manager with configuration for isolcpus and SMT; Multiple interfaces; NUMA topology; Device Plugin |
```

**Review by tomkivlin — CHANGES_REQUESTED** [review]

File: doc/ref_arch/kubernetes/chapters/chapter03.md [hunk 5]
```diff
@@ -206,6 +206,23 @@ which provides the isolation of Operating System kernels.
 The architecture must support a way to isolate the compute resources of the
 infrastructure itself from the workloads compute resources.
 
+
+<a name="3.2.1.8"></a>
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter03.md:210]: Previous section is 3.2.1.5
```suggestion
<a name="3.2.1.6"></a>
```

File: doc/ref_arch/kubernetes/chapters/chapter03.md [hunk 6]
```diff
@@ -206,6 +206,23 @@ which provides the isolation of Operating System kernels.
...
+#### 3.2.1.8 Scheduling Pods with Non-resilient Applications
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter03.md:211]: Previous section is 3.2.1.5
```suggestion
#### 3.2.1.6 Scheduling Pods with Non-resilient Applications
```

**Comment by pgoyal01** [comment]
@petorre The ToC needs to be changed with the new 3.2.1.6 (originally, 3.2.1.8)

**Comment by petorre** [comment]
> 
> 
> @petorre The ToC needs to be changed with the new 3.2.1.6 (originally, 3.2.1.8)

Please see #2145 "Reorder 3.2.1.x sections and TOC".

**Review by pgoyal01 — APPROVED** [review]

**Review by tomkivlin — APPROVED** [review]

**Review by rgstori — APPROVED** [review]

**Comment by tomkivlin** [comment]
@petorre could you resolve the conflicts and then I can merge. Thanks!

**Comment by petorre** [comment]
> @petorre could you resolve the conflicts and then I can merge. Thanks!

Done.

**Pull request merged**